datahub-web-react/src/app/identity/user/SelectRole.tsx

```diff
@@ -29,29 +32,27 @@ export default function SelectRole({ user, userRoleUrn, selectRoleOptions, refet
     selectRoleOptions.forEach((role) => {
         rolesMap.set(role.urn, role);
     });
-    const allSelectRoleOptions = [{ urn: '', name: NO_ROLE_TEXT }, ...selectRoleOptions];
-    const selectOptions = () =>
-        allSelectRoleOptions.map((role) => {
-            return (
-                <Select.Option value={role.urn}>
-                    <RoleIcon>{mapRoleIcon(role.name)}</RoleIcon>
-                    {role.name}
-                </Select.Option>
-            );
-        });
+    const allSelectRoleOptions = [{ urn: NO_ROLE_URN, name: NO_ROLE_TEXT }, ...selectRoleOptions];
+    const selectOptions = allSelectRoleOptions.map((role) => {
+        return (
+            <Select.Option key={role.urn} value={role.urn}>
+                <RoleIcon>{mapRoleIcon(role.name)}</RoleIcon>
+                {role.name}
+            </Select.Option>
+        );
+    });
 
-    const initialRole = rolesMap.get(userRoleUrn) as DataHubRole;
-    const [currentRole, setCurrentRole] = useState<DataHubRole | undefined>(initialRole);
+    const defaultRoleUrn = userRoleUrn || NO_ROLE_URN;
+    const [currentRole, setCurrentRole] = useState<string>(defaultRoleUrn);
```

Comment:
Minor: We should probably call this currentRoleUrn

Reply:
sounds good